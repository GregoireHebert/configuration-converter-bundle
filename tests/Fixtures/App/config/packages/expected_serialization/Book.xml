<?xml version="1.0"?>
<serializer xmlns="http://symfony.com/schema/dic/serializer-mapping" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://symfony.com/schema/dic/serializer-mapping https://symfony.com/schema/dic/serializer-mapping/serializer-mapping-1.0.xsd">
  <class name="ConfigurationConverter\Test\Fixtures\App\src\Entity\Book">
    <attribute name="name">
      <group>read</group>
      <group>write</group>
      <group>query</group>
    </attribute>
    <attribute name="author">
      <group>read</group>
      <group>mutation</group>
    </attribute>
  </class>
</serializer>
