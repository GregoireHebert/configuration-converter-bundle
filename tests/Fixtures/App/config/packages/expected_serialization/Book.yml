ConfigurationConverter\Test\Fixtures\App\src\Entity\Book:
    attributes:
        name:
            groups: [read, write, query]
        author:
            groups: [read, mutation]
