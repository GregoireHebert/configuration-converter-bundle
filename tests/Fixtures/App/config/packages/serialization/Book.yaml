ConfigurationConverter\Test\Fixtures\Entity\Book:
    attributes:
        name:
            groups: [read, write, query]
        author:
            groups: [read, mutation]
