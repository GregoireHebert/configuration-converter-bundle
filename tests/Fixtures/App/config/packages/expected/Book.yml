ConfigurationConverter\Test\Fixtures\App\src\Entity\Book:
    shortName: customShortname
    description: 'my description'
    iri: 'http://schema.org/Book'
    attributes:
        normalization_context:
            groups:
                - read
        denormalization_context:
            groups:
                - write
        deprecation_reason: 'Create a Book instead'
    graphql:
        query:
            normalization_context:
                groups:
                    - query
        create:
            normalization_context:
                groups:
                    - query
            denormalization_context:
                groups:
                    - mutation
        delete: null
    itemOperations:
        get:
            method: GET
            path: '/grimoire/{id}'
            requirements:
                id: \d+
            defaults:
                color: brown
            options:
                my_option: my_option_value
            schemes:
                - https
            host: '{subdomain}.api-platform.com'
        put:
            method: PUT
            path: '/grimoire/{id}/update'
            hydra_context:
                foo: bar
        post_publication:
            method: PUT
            path: /grimoire
            deprecation_reason: 'Retrieve a Book instead'
            sunset: 01/01/2020
    collectionOperations:
        get:
            attribute:
                filters:
                    - customShortname.GroupFilter
                    - customShortname.PropertyFilter
                    - customShortname.SearchFilter
                    - customShortname.OrderFilter
        custom:
            method: GET
            path: /MyRoute
        post: null
    properties:
        id:
            description: 'identifiant unique'
            iri: 'http://schema.org/id'
            readable: true
            writable: true
            readableLink: true
            writableLink: true
            required: true
            identifier: true
        name:
            readable: true
            writable: true
        author:
            readable: true
            subresource:
                resourceClass: ConfigurationConverter\Test\Fixtures\App\src\Entity\Dummy
                maxDepth: 1
        scribus:
            attributes:
                deprecation_reason: 'Use the author property instead'
